### DISCUSSION TITLE
Setting Index Page Without Modifying `baseUrl` When Using Multi-Instance Setup with Defined `routeBasePath`

### DISCUSSION THREAD

**Discussion by kmlbgn** [facebook/docusaurus#9926] [Q&A] [answered]
Hello,

I'm currently working with a multi-instance setup in Docusaurus, containing four distinct structures (let's call them i1, ..., i4) all listed as items on the top navbar. 

Despite following the documentation to configure one instance as the default (without an ID) in the `plugins` section, or by using at least one of the instances as a `presets` docs instance in the config, if all instances have a `routeBasePath` different than "/", I encounter a "PageNotFound" error when setting the `baseUrl` to "/". 

This outcome seems logical, but what if I wish to keep the entire site prefix-free (using `baseUrl` "/") but have it's index page set to the default instance's index page? Is there a strategy to achieve this without resorting to a custom index.js in `src/pages` for redirection?

The only workaround I can think of is to rewrite the favicon and the 'home' icon href to point to the default istance, which would make `baseUrl` "/" only reachable by manual navigation.

**Context: I want all my instances to have a defined `routeBasePath` because I need all my links to have this format `/instance/slug`, instead of one instance being only `/slug` which is impractical with my current setup.** 

Thank you for your assistance.

**Comment by slorber** [comment]
If all your plugins have a base url, then what you consider as the "default instance" has one too: let's say it's `/defaultInstance/`.

So the root of this plugin is `/defaultInstance/`, and you want to also access it through `/`. That's technically possible (through a reverse proxy/CDN option/edge middleware) but not recommended for SEO reasons because you'd have 2 URLs serving the same page now. Although we define a canonical URL metadata in the pages, it remains preferable to serve them all from a single URL instead of multiple ones.

My suggestion would be to use your proxy/CDN/middleware config to add a server redirect from `/` to `/defaultInstance/`, so that only `/defaultInstance/` serves a response.

**Reply by kmlbgn** [reply]
Actually I don't want to access any of the plugins instances via "/", and nothing should exist at "/". 
However I want `baseUrl` to be set to "/" to stay prefix-free, but because `baseUrl` sets the href of little "home" icon in the breadcrumbs navigation, it leads to 404. 

My current workaround is using this simple `index.js` which isn't bad but I was wondering if there was a dedicated solution. I think the problem is that because `baseUrl` is used to setup links such as the home icon, it forces you to serve something at this level. 

```
import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';

function RedirectToLearn() {
  const history = useHistory();

  useEffect(() => {
    history.push('/learn/');
  }, [history]);

  return null; // or a loader component if you prefer
}

export default RedirectToLearn;
```

**Reply by slorber** [reply, marked as answer]
Hey
It's becoming difficult to follow for me.

Can you be more concrete and give me the an actual list of plugins, and which URLs should be accessed?

Maybe what we miss is the lack of customization options for the breadcrumb home icon. You can swizzle it in the meantime.

https://github.com/facebook/docusaurus/issues/6953

**Reply by kmlbgn** [reply]
That issue is exactly what i'm looking for. Thanks ! 

Basically my problem is that i want my entire documentation to exist like this:  `/baseUrl/instance-name/...`
no page should be served at `/baseUrl/`. And that breadcrumb icon was pointing to it.
